kyo-kernel/shared/src/main/scala/kyo/kernel/internal/CanLift.scala

```diff
@@ -0,0 +1,64 @@
+package kyo.kernel.internal
+
+import kyo.<
+import scala.annotation.implicitNotFound
+import scala.quoted.*
+import scala.util.NotGiven
+
+/** CanLift is a "soft" constraint that indicates a type should not contain nested effect computations (A < S), or A is not a module from
+  * kyo (like Abort.type).
+  *
+  * This constraint helps:
+  *   - prevent accidental nesting of effects that would require flattening, but cannot be strictly enforced in all generic contexts,
+  *   - prevent calling combinators from (A < S) on modules, like Abort.foldAbort.
+  *
+  * @tparam A
+  *   The type to check for nested effects
+  */
+@implicitNotFound("""
+Type '${A}' may contain a nested effect computation.
+
+This usually means you have a value of type 'X < S' where you need a plain value.
+To fix this, you can:
+
+1. Call .flatten to combine the nested effects:
+     (x: (Int < S1) < S2).flatten  // Result: Int < (S1 & S2)
+
+2. Split the computation into multiple statements:
+     val x: Int < S1 = computeFirst()
+     val y: Int < S2 = useResult(x)
+""")
```

Comment:
@hearnadam something like that?

```suggestion
**Type '${A}' may contain a nested effect computation.**
This usually means you have a value of type `X < S1 < S2` (i.e. `(X < S1) < S2`) where a plain value `X < S` is expected.

This often happens due to *type inference*: some effect computations are nested when chaining operations, and Scala infers a value with nested effects instead of merging them.

To fix this, you can:

1. Call `.flatten` to merge the nested effects:
```
val x: (Int < S1) < S2 = ...
val y: Int < (S1 & S2) = x.flatten
```

   This collapses the nested effect layers into a single computation with a combined effect set.

2. Split the computation into multiple statements:
   Breaking the code into smaller expressions helps Scala infer the correct types incrementally, avoiding nested effects.
```
val x: Int < S1 = computeFirst()
val y: Int < S2 = useResult(x)
```
""")
```